test/integration/objectid.test.js

```diff
@@ -1,92 +1,43 @@
 'use strict';
 var test = require('./shared').assert;
 const { expect } = require('chai');
-var setupDatabase = require('./shared').setupDatabase;
 const { ObjectId } = require('../../src');
-const { clearInterval, setInterval } = require('timers');
 const { sleep } = require('../tools/utils');
 
 describe('ObjectId', function () {
-  before(function () {
-    return setupDatabase(this.configuration);
+  let client;
+  beforeEach(async function () {
+    client = this.configuration.newClient();
   });
 
-  it('shouldCorrectlyGenerateObjectId', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
+  afterEach(async function () {
+    await client.close();
+  });
 
-    test: function (done) {
-      var configuration = this.configuration;
-      var client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
-      var db = client.db(configuration.db);
-      var number_of_tests_done = 0;
-
-      var collection = db.collection('test_object_id_generation.data');
-      // Insert test documents (creates collections and test fetch by query)
-      collection.insertMany(
-        [{ name: 'Fred', age: 42 }],
-        { writeConcern: { w: 1 } },
-        function (err, r) {
-          expect(r).property('insertedCount').to.equal(1);
-
-          const id = r.insertedIds[0];
-          expect(id.toHexString().length).to.equal(24);
-          // Locate the first document inserted
-          collection.findOne({ name: 'Fred' }, function (err, document) {
-            expect(err).to.not.exist;
-            expect(id.toHexString()).to.equal(document._id.toHexString());
-            number_of_tests_done++;
-          });
-        }
-      );
-
-      // Insert another test document and collect using ObjectId
-      collection.insert({ name: 'Pat', age: 21 }, { writeConcern: { w: 1 } }, function (err, r) {
-        expect(r).property('insertedCount').to.equal(1);
-
-        const id = r.insertedIds[0];
-        expect(id.toHexString().length).to.equal(24);
-
-        // Locate the first document inserted
-        collection.findOne(id, function (err, document) {
-          expect(err).to.not.exist;
-          expect(id.toHexString()).to.equal(document._id.toHexString());
-          number_of_tests_done++;
-        });
-      });
+  it('generates new ObjectId for documents without _id property', async function () {
```

Comment:
It failed at first b/c of the insert usage, but then I also cleaned it up cus the setInterval logic was too interesting